src/medCore/views/medAbstractLayeredView.cpp

```diff
@@ -463,16 +463,20 @@ void medAbstractLayeredView::write(QString& path)
 
         //getting a working file extension
         //1. find suitable writers
+        QList<QString> allWriters = medAbstractDataFactory::instance()->writers();
         QHash<QString, dtkAbstractDataWriter*> possibleWriters=medDataManager::instance()->getPossibleWriters(layerData(i));
 
-
         //2. use these writers to get a suitable file extension
         QString fileExtension;
-        foreach(QString key,possibleWriters.keys())
-        {
-            if(!possibleWriters[key]->supportedFileExtensions().isEmpty())
+        // we use allWriters as the list of keys to make sure we traverse possibleWriters
+        // in the order specified by the writers priorities.
+        foreach(QString type, allWriters) {
+            if (!possibleWriters.contains(type))
+                continue;
```

Comment:
I agree with @fcollot , the way "continue" is used here is not really clear. You can just do a `if(positive case){do something}` :)